configurd.go

```diff
@@ -29,18 +44,37 @@ type Configurd struct {
 	Namespaces []Namespace
 }
 
-func New(dir string) (Configurd, error) {
-	c := Configurd{}
+type serviceLoader func(string) (Service, error)
+
+type Logger interface {
+	Printf(format string, args ...interface{})
+	Fatalf(format string, args ...interface{})
+}
+
+func relativePath(path, basePath string) string {
+	return strings.Replace(path, basePath+"/", "", -1)
+}
 
-	serviceDir := path.Join(dir, defaultServiceDir)
+func New(log Logger, basePath string) (Configurd, error) {
+	log.Printf("Parsing configuration files")
+	log.Printf("Entering dir: %q", basePath)
+
+	c := Configurd{}
+	serviceDir := path.Join(basePath, defaultServiceDir)
 	err := filepath.Walk(serviceDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
 		if !info.IsDir() && (strings.HasSuffix(path, defaultConfigFileExt)) {
 			file, err := os.Open(path)
 			if err != nil {
 				return err
 			}
 			defer file.Close()
-			svc, err := LoadService(file)
+
+			var name = info.Name()[0 : len(info.Name())-len(filepath.Ext(info.Name()))]
```

Comment:
name := (stay consistent)

Reply:
or a nicer way would be name := strings.TrimSuffix(info.Name(), filepath.Ext(info.Name()))

Reply:
or even better would be name := strings.TrimSuffix(info.Name(), defaultConfigFileExt) since it's checked